spec/controllers/drivers_spec.cr

```diff
@@ -90,34 +95,29 @@ module PlaceOS::Api
           )
 
           response.success?.should be_true
+          clear_tables
         end
-      end
 
-      # TODO: awaiting addition of `core` to test environment
-      pending "POST /:id/recompile" do
-        driver = Model::Generator.driver.save!
-        Model::Generator.module(driver: driver).save!
-
-        response = curl(
-          method: "GET",
-          path: "#{base}#{driver.id.not_nil!}/compiled",
-          headers: authorization_header.merge({"Content-Type" => "application/json"}),
-        )
-
-        response.success?.should be_true
-
-        path = "#{base}#{driver.id.not_nil!}/recompile"
-        _result = curl(
-          method: "POST",
-          path: path,
-          headers: authorization_header.merge({"Content-Type" => "application/json"}),
-        )
+        it "POST /:id/recompile" do
+          driver, _, _, _ = setup_system
+
+          response = curl(
+            method: "POST",
+            path: "#{base}#{driver.id.not_nil!}/recompile",
+            headers: authorization_header.merge({"Content-Type" => "application/json"}),
+          )
+
+          response.success?.should be_true
+          updated = Model::Driver.from_trusted_json(response.body)
+          updated.commit.starts_with?("RECOMPILE").should be_false
+          clear_tables
+        end
       end
 
       describe "scopes" do
         before_each do
-          HttpMocks.etcd_range
           HttpMocks.core_compiled
+          # HttpMocks.reset
```

Comment:
I assume we want to enable/disable these mocks depending on whether they depend on core or not.